src/codemodder/codemods/sonar.py

```diff
@@ -1,16 +1,45 @@
 from pathlib import Path
+from codemodder.codemods.base_codemod import Metadata, Reference
 from codemodder.codemods.base_detector import BaseDetector
+from codemodder.codemods.base_transformer import BaseTransformerPipeline
 from codemodder.context import CodemodExecutionContext
 from codemodder.result import ResultSet
 from codemodder.sonar_results import SonarResultSet
-from core_codemods.api.core_codemod import SASTCodemod
+from core_codemods.api.core_codemod import CoreCodemod, SASTCodemod
 
 
 class SonarCodemod(SASTCodemod):
     @property
     def origin(self):
         return "sonar"
 
+    @classmethod
+    def from_core_codemod(
+        cls,
+        name: str,
+        other: CoreCodemod,
+        rules: list[str],
+        transformer: BaseTransformerPipeline | None = None,
+        new_references: list[Reference] | None = None,
+    ):  # pylint: disable=too-many-arguments
```

Comment:
Not for this PR but I think we should just reconfigure or disable this pylint rule globally.